### PR TITLE
Use correct filename when using sidecar injection

### PR DESCRIPTION
**Pull request by selmanj**
Existing install script uses an invalid path name, leading to 503 errors when attempting to access a service.

### CODE REVIEW

File: install/gcp/deployment_manager/istio-cluster.jinja [hunk 1]
```diff
@@ -97,7 +97,7 @@ resources:
           {% endif %}
 
           {% if  properties['enableAutomaticSidecarInjection'] %}
-            kubectl apply -f install/kubernetes/istio-sidecar-injector.yaml
+            kubectl apply -f install/kubernetes/istio-initializer.yaml
```
Comment (ayj) [inline install/gcp/deployment_manager/istio-cluster.jinja]: The initializer was removed in 0.5.0 and replaced with the sidecar injector.

**Comment by selmanj** [comment]
@ayj Good to know - currently 0.5.0 isn't an available option but I understand it was just released. When I add it to the list I'll update the script to install properly.

Also, note that this file isn't currently covered by any e2e testing, so the failure should be unrelated.

**Comment by andraxylia** [comment]
@ayj I am confused, should this be merged?

**Comment by ayj** [comment]
Looks like GCP deployment_manager setup is pinned to a specific Istio release. See https://github.com/istio/istio/blob/master/install/gcp/deployment_manager/istio-cluster.yaml#L20.

**Comment by ayj** [comment]
> @ayj I am confused, should this be merged?

GCP install files for sidecar injection will be updated in a follow-up PR (see https://github.com/istio/istio/issues/3047)

**Comment by selmanj** [comment]
Also as a reminder, this template file isn't executed in any test path - any test failures are unrelated to this change.

**Comment by ldemailly** [comment]
istio/install/gcp/deployment_manager/istio-cluster.yaml should really not hard code the istio version (I think I mentioned that when it first was added) - what's the eta on that ?

**Comment by selmanj** [comment]
@ldemailly Not currently planned. 

I'm not sure it's a good idea; for example, I'm working on updating the install script to support 0.5.0. The automatic sidecar injection process is different (now it requires setting up a k8s admission webhook which involves cert creation, etc). If we had updated the list of supported versions automatically without remembering to adjust the install script, then it would have been broken.

Right now I've tried to emphasize that this script is owned by the community (currently me and @salrashid123) and not maintained by the core istio team. Happy to discuss if the istio team wants to maintain this method of installation as a fully supported installation method.

**Comment by ldemailly** [comment]
we should pretty soon take installation backward compatibility more seriously but the best way to keep it in sync is to add a test and fix things as they get changed

can we make sure it doesn't lag too much, 0.3.0 is pretty old by now

**Comment by selmanj** [comment]
I'll look at adding a test binary that we can get hooked up to the CI process (although it will likely be brittle as it will involve using real prod credentials to talk to GCP). 

As for supported versions, our policy has been to support the three most recent releases - once 0.5.0 support is added we'll drop 0.2.12 as an option.

**Comment by selmanj** [comment]
Also note I'd prefer not to block this change on improvements in the script; currently installing the sidecar on versions earlier than 0.5.0 is broken until this change is pulled in.

**Comment by ldemailly** [comment]
I don't think we can guarantee master will work forever with 0.3.0 binaries - so the installation should match the release - why not reference https://github.com/istio/istio/blob/0.4.0/install/gcp/deployment_manager/istio-cluster.jinja in the doc for all pre-0.5 releases?

**Comment by selmanj** [comment]
@ldemailly That's a fair concern; the way we've been trying to support that currently is by having a version dropdown/field in the schema. But we could move to tagging specific versions and linking to them from the quick start documentation.

I can open a related issue, but I'd prefer not to block this change on that work; this is a bug fix that is currently impacting users who opt for sidecar injection.

**Comment by andraxylia** [comment]
failure in e2e-pilot is caused by routing rules with websockets #3220, unrelated to this PR.

**Comment by andraxylia** [comment]
@ayj @ldemailly I see lots of comments and it is not clear if this PR should be merged or not. Please apply hold label if this should not be merged.

**Comment by rshriram** [comment]
please sync with master.

**Comment by ldemailly** [comment]
This can be merged as there isn't a readily available better solution

**Pull request merged**